rust/vectordb/src/table.rs

```diff
@@ -227,46 +327,22 @@ impl Table {
                     message: e.to_string(),
                 },
             })?;
-        Ok(Table {
+        Ok(NativeTable {
             name: name.to_string(),
             uri: uri.to_string(),
-            dataset: Arc::new(dataset),
+            dataset: Arc::new(Mutex::new(dataset)),
```

Comment:
Does napi require `NativeTable` to be `Sync`?  Why does this need to be in a `Mutex`?

I think it's fine, I'm mostly just curious.

Reply:
Yes, so `napi` needs `dyn Table` to be `Sync`, which makes it hard to have mutable function `fn foo(&mut self, ...)`